src/IdRegistry.sol

```diff
@@ -318,4 +318,25 @@ contract IdRegistry is TrustedCaller, Signatures, Pausable, EIP712, Nonces {
             sig
         );
     }
+
+    /*//////////////////////////////////////////////////////////////
+                             EXTERNAL VIEW HELPERS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Verify that a signature was produced by the custody address that owns an fid.
+     *
+     * @param signer   The address to check the signature of.
+     * @param fid      The fid to check the signature of.
+     * @param digest   The digest that was signed.
+     * @param sig      The signature to check.
+     */
+    function verifyFidCustodyAddress(
+        address signer,
+        uint256 fid,
+        bytes32 digest,
+        bytes calldata sig
+    ) external view returns (bool is_valid) {
+        is_valid = idOf[signer] == fid && SignatureChecker.isValidSignatureNow(signer, digest, sig);
```

Comment:
should we just look up the address here instead of asking the user to provide it? is there a benefit to asking?

Reply:
https://github.com/farcasterxyz/contracts/blob/ccb80bb8a7edaa713552d4f81dc298856c3a1b4c/src/IdRegistry.sol#L102-L107

There is no mapping from fid to owner address.

There is only owner->fid, and fid->recoveryAddress.


```
    function _unsafeRegister(address to, address recovery) internal whenNotPaused returns (uint256 fid) {
        /* Revert if the target(to) has an fid */
        if (idOf[to] != 0) revert HasId();

        /* Safety: idCounter won't realistically overflow. */
        unchecked {
            /* Incrementing before assignment ensures that no one gets the 0 fid. */
            fid = ++idCounter;
        }

        /* Perf: Save 29 gas by avoiding to == address(0) check since 0x0 can only register 1 fid */
        idOf[to] = fid;
        recoveryOf[fid] = recovery;
    }
```

The contract assumes an address can only have 1 fid. There is no way to resolve a fid to it's owner address from the contract.

Would require a pretty significant change to the contract to add fid->owner resolution. You probably want to just make it a NFT contract with a 1 nft per address limitation at that point.

Reply:
One alternative is to not as for a fid, and return the fid is sig is valid or throw an error. 

I think this alternative should be a separate view func though.

The main point of the `verifyFidSignature` is to make the developer experience easier if anyone needs to verify a fid signature off-chain. This is important considering the erc-1271 support that was added.


We could also add a `recoverFid` that doesn't take a fid (still need custodyAddress). This could also be useful to have.